src/main/cpp/src/cast_string.cu

```diff
@@ -140,58 +133,24 @@ __global__ void string_to_integer_kernel(T* out,
 
           thread_val *= 10;
         }
-        T original_val = thread_val;
         if (is_signed_type && sign < 0) {
-          thread_val -= (chr - '0');
-          if (debug)
-            printf(
-              "%d %d - original_val %d(unsigned %u) thread_val %d(unsigned %u), (thread_val > "
-              "orig_val)==%s, (int comparison)==%s \n",
-              threadIdx.x,
-              blockIdx.x,
-              original_val,
-              original_val,
-              thread_val,
-              thread_val,
-              (thread_val > original_val ? "true" : "false"),
-              (static_cast<int32_t>(thread_val) > static_cast<int32_t>(original_val) ? "true"
-                                                                                     : "false"));
-          if ((T)thread_val > (T)original_val) {
-            if (debug)
-              printf("%d %d - overflow because %d is > %d\n",
-                     threadIdx.x,
-                     blockIdx.x,
-                     thread_val,
-                     original_val);
+          auto const c       = chr - '0';
+          auto const min_val = std::numeric_limits<T>::min() + c;
+          if (min_val > thread_val) {
```

Comment:
Nit: The similar code above uses `thread_val < miin_val` and t would be nice to keep it consistent